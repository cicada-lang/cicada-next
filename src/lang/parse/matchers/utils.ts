import * as Exps from "../../exp"

export function piBindingtoFnBindingFrom(binding: Exps.PiBinding): Exps.FnBinding {
  switch (binding.kind) {
    case "PiBindingNameless": {
      return Exps.FnBindingName("_")
    }

    case "PiBindingNamed": {
      return Exps.FnBindingName(binding.name)
    }

    case "PiBindingImplicit": {
      return Exps.FnBindingImplicit(binding.name)
    }
  }
}
