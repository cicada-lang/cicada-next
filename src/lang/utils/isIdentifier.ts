const identifierRegExp = new RegExp("^([-_\\p{Letter}][-_\\p{Letter}0-9]*)$", "u")

export function isIdentifier(text: string): boolean {
  return identifierRegExp.test(text)
}
