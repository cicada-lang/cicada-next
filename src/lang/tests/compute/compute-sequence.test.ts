import { expect, test } from "vitest"
import { runCode } from "../utils"

test("compute Sequence -- single return", async () => {
  const output = await runCode(`

compute {
  return sole
}

`)

  expect(output).toMatchSnapshot()
})

test("compute Sequence -- Let", async () => {
  const output = await runCode(`

compute {
  let x = sole
  return x
}

`)

  expect(output).toMatchSnapshot()
})

test("compute Sequence -- LetThe", async () => {
  const output = await runCode(`

compute {
  let x: Trivial = sole
  return x
}

`)

  expect(output).toMatchSnapshot()
})

test("compute Sequence -- Check", async () => {
  const output = await runCode(`

compute {
  check sole: Trivial
  return sole
}

`)

  expect(output).toMatchSnapshot()
})
