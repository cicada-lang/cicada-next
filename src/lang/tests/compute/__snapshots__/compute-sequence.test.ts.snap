// Vitest Snapshot v1, https://bun:test.dev/guide/snapshot.html

exports[`compute Sequence -- Check 1`] = `"sole: Trivial"`;

exports[`compute Sequence -- Let 1`] = `"sole: Trivial"`;

exports[`compute Sequence -- LetThe 1`] = `"sole: Trivial"`;

exports[`compute Sequence -- single return 1`] = `"sole: Trivial"`;
